sigar_linux.go

```diff
@@ -562,7 +562,29 @@ func (self *DiskList) Get() error {
 }
 
 func (self *ProcessList) Get() error {
-	return notImplemented()
+	pids := ProcList{}
+	err := pids.Get()
+	if err != nil {
+		return err
+	}
+
+	processes := make([]Process, 0, len(pids.List))
+	for _, pid := range pids.List {
+		var process Process
+
+		// Gather each composed struct, ignoring any errors.
```

Comment:
I wish we could do something better here, but there is no logger, and immediately returning an error would prevent us from gathering any other structs which could work fine. Basically, we'd throw out all the results due to an issue with one of them.

On the other hand, the contract could be that results in `Process` are _not_ undefined when an error returns, and we could return only the first error. Going down this path, ideally I think we would want an `IsFatal` interface wrapping those errors (similar to <a href="https://golang.org/pkg/os/#IsPermission">IsPermission</a>), allowing us to effectively say "I had a problem with one collector, but mostly things are okay" vs "I had a fatal error and couldn't return anything useful". Its additional work, but I'm not sure we would effectively use the information upstream, particularly since platform support for various fields is spotty anyway, and we already have to deal with missing information and zero-valued struct members even when an error doesn't occur.